test/go/ferry_test.go

```diff
@@ -37,6 +32,31 @@ func (t *FerryTestSuite) TestReadOnlyDatabaseFailsInitialization() {
 	ferry = testhelpers.NewTestFerry().Ferry
 	err = ferry.Initialize()
 	t.Require().Nil(err)
+
+	_, err = t.Ferry.TargetDB.Exec("SET GLOBAL read_only = OFF")
+	t.Require().Nil(err)
+}
+
+func (t *FerryTestSuite) TestSourceDatabaseWithForeignKeyConstraintFailsInitialization() {
```

Comment:
Can you add a test for `SkipForeignKeyConstraintsCheck`

Reply:
👍 Added a set of assertions in the same test.